packages/manager/src/features/VPC/VPCCreate/VPCCreate.tsx

```diff
@@ -1,11 +1,243 @@
 import { DocumentTitleSegment } from 'src/components/DocumentTitle';
+import { LandingHeader } from 'src/components/LandingHeader';
+import { Paper } from 'src/components/Paper';
+import { ActionsPanel } from 'src/components/ActionsPanel/ActionsPanel';
+import { handleAPIErrors } from 'src/utilities/formikErrorUtils';
+import { SubnetFieldState, validateSubnets } from 'src/utilities/subnets';
+import { MultipleSubnetInput } from './MultipleSubnetInput';
+import {
+  StyledBodyTypography,
+  StyledHeaderTypography,
+} from './VPCCreate.styles';
 
 const VPCCreate = () => {
+  const theme = useTheme();
+  const history = useHistory();
+  const { data: profile } = useProfile();
+  const { data: grants } = useGrants();
+  const { data: regions } = useRegionsQuery();
+  const { isLoading, mutateAsync: createVPC } = useCreateVPCMutation();
+  const [subnetErrorsFromAPI, setSubnetErrorsFromAPI] = React.useState<
+    APIError[]
+  >();
+
+  const disabled = profile?.restricted && !grants?.global.add_vpcs;
+
+  const createSubnetsPayload = () => {
+    const subnetPayloads: CreateSubnetPayload[] = [];
+
+    for (const subnetState of values.subnets) {
+      const { label, ip } = subnetState;
+      if (ip.ipv4 || label) {
+        subnetPayloads.push({ label: label, ipv4: ip.ipv4 });
+      }
+    }
+
+    return subnetPayloads;
+  };
+
+  const validateVPCSubnets = () => {
+    const validatedSubnets = validateSubnets(values.subnets, {
+      labelError:
+        'Label is required. Must only be ASCII letters, numbers, and dashes',
+      ipv4Error: 'The IPv4 range must be in CIDR format',
+    });
+
+    if (
+      validatedSubnets.some(
+        (subnet) => subnet.labelError || subnet.ip.ipv4Error
+      )
+    ) {
+      setFieldValue('subnets', validatedSubnets);
+      return false;
+    } else {
+      setFieldValue(
+        'subnets',
+        validatedSubnets.map((subnet) => {
+          delete subnet.labelError;
+          delete subnet.ip.ipv4Error;
+          return { ...subnet };
+        })
+      );
+      return true;
+    }
+  };
+
+  const onCreateVPC = async () => {
+    if (!validateVPCSubnets()) {
+      return;
+    }
+
+    setSubmitting(true);
+
+    const subnetsPayload = createSubnetsPayload();
+
+    const createVPCPayload: CreateVPCPayload = {
+      ...values,
+      subnets: subnetsPayload,
+    };
+
+    try {
+      const response = await createVPC(createVPCPayload);
+      history.push(`/vpc/${response.id}`);
+    } catch (errors) {
+      const apiSubnetErrors = errors.filter(
+        (error: APIError) => error.field === 'subnets'
+      );
+      if (apiSubnetErrors) {
+        setSubnetErrorsFromAPI(apiSubnetErrors);
+      }
+      handleAPIErrors(errors, setFieldError);
+    }
+
+    setSubmitting(false);
+  };
+
+  const {
+    values,
+    errors,
+    setFieldValue,
+    setFieldError,
+    setSubmitting,
+    handleChange,
+    handleSubmit,
+  } = useFormik({
```

Comment:
I think it is best to try to keep the Formik values in the shape the API would expect. Is this possible?

```suggestion
  } = useFormik<CreateVPCPayload>({
```

Reply:
Imo it's possible but potentially weird, due to the validation checks on subnets. In the other [validation version here](https://github.com/coliu-akamai/manager/pull/1), I at one point had the Formik values in the shape of the API data, but when I deleted subnets, the errors shown on the UI would shift and no longer correspond to the same subnets. One idea I had to get around that would be to clear subnet errors any time there's a change to the subnets (but to clear all subnet errors if I change a single subnet feels a bit strange)